### ISSUE TITLE
απορίες  2ο εργαστήριο

### ISSUE THREAD

**Issue by dzerkes** [slp-ntua/patreco-lab-support-19-20#11]
Γειά σας, θα ήθελα να κάνω κάποιες διευκρινιστικές ερωτήσεις για την εργαστηριακή άσκηση, μιας που χάθηκε το εργαστήριο σήμερα για να είμαστε έτοιμοι τη Δευτέρα.

Τι εννοείτε στο Βήμα 4 , να αναπαραστήσουμε γραφικά τη συσχέτιση των MFSC και MFCC σε διαφορετικά διαγράμματα για την κάθε εκφώνηση. Να κάνουμε κάποιο spectogram ή κάτι άλλο?

Επίσης, στο βήμα 5 , ας υποθέσουμε ότι έχουμε μια εκφώνηση που χει 13 mfcc χαρακτηριστικ'α και πίνακα 13x3000(ή όσο είναι η 2η διάσταση αφού ενώσουμε τα mfcc-deltas-deltas-deltas). Θέλετε ουσιαστικά να το μετατρέψουμε σε ένα πίνακα 13χ2 όπου η μια στήλη θα είναι ο μέσος του κάθε mfcc και η άλλη η τυπική απόκλιση του  kαι στη συνέχεια αυτά θα τα κάνουμε για παράδειγμα flatten όπως σε εικόνες με pixel?

Αυτά έχουμε κάνει πάνω κάτω και στο 3d plot δεν υπήρχε κάποιος ιδιαίτερος διαχωρισμός και οι ταξινομητές βγάζαν κακό accuracy. 

Απλά θέλαμε να δούμε δηλαδή αν έχουμε καταλάβει κάτι λάθος ή είναι οκ.

**Comment by efthymisgeo** [comment]
Σχετικά με το Βήμα 4 παραπέμπουμε στα ακόλουθα:
- [https://towardsdatascience.com/better-heatmaps-and-correlation-matrix-plots-in-python-41445d0f2bec]
- [http://benalexkeen.com/correlation-in-python/]
Ουσιαστικά θέλουμε τους λεγόμενους **πίνακες συσχέτισης**. Πιο πολλά μπορείτε να βρείτε και εδώ
- [https://towardsdatascience.com/let-us-understand-the-correlation-matrix-and-covariance-matrix-d42e6b643c22]

Σε ότι αφορά το βήμα 5 ναι θέλουμε μία διανυσματική αναπαράσταση για κάθε εκφώνηση. Συνεπώς η διαδικασία που προτείνετε είναι ορθή.

**Comment by dzerkes** [comment]
οκ ευχαριστώ πολύ!

**Comment by panosRaptis** [comment]
Καλησπέρα σας,

Προκειμένου να δω εάν έχω καταλάβει καλά, για κάθε γραμμή του πίνακα διάστασης 13 x 3*(#windows), ας πούμε πίνακας Α, πηγαίνουμε σε έναν πίνακα Β διάστασης 13*2 με 1η κολώνα το mean, 2η το var της αντίστοιχης γραμμής του πίνακα Α? Και έπειτα δημιουργούμε το διάνυσμα [Β[0][0], Β[0][1], Β[1][0] ...] με 26 στοιχεία?

Ευχαριστώ!

**Comment by georgepar** [comment]
> Καλησπέρα σας,
> 
> Προκειμένου να δω εάν έχω καταλάβει καλά, για κάθε γραμμή του πίνακα διάστασης 13 x 3*(#windows), ας πούμε πίνακας Α, πηγαίνουμε σε έναν πίνακα Β διάστασης 13*2 με 1η κολώνα το mean, 2η το var της αντίστοιχης γραμμής του πίνακα Α? Και έπειτα δημιουργούμε το διάνυσμα [Β[0][0], Β[0][1], Β[1][0] ...] με 26 στοιχεία?
> 
> Ευχαριστώ!

Όχι. Ο πίνακας των MFCCs αρχικά είναι 13 x (#frames). Μετά υπολογίζετε deltas + deltadeltas και πάτε σε ένα πίνακα 3 x 13 x (#frames) τον οποίο το μετατρέπετε σε 39 x (#frames).
Μετά μπορείτε να πάρετε τη μέση τιμή και την τυπική απόκλιση πάνω στο χρόνο (frames) για να καταλήξετε με 2 διανύσματα μήκους 39.
Στους ταξινομητές σας θα βάλετε ένα διάνυσμα με 78 στοιχεία
